dali/kernels/erase/erase_gpu_test.cu

```diff
@@ -0,0 +1,399 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <cmath>
+#include <complex>
+#include <tuple>
+#include <vector>
+
+#include "dali/kernels/common/utils.h"
+#include "dali/kernels/erase/erase_gpu.h"
+#include "dali/kernels/scratch.h"
+#include "dali/pipeline/data/tensor_list.h"
+#include "dali/test/tensor_test_utils.h"
+#include "dali/test/test_tensors.h"
+
+#include "dali/kernels/erase/erase_cpu.h"
+
+#include "dali/core/cuda_event.h"
+
+namespace dali {
+namespace kernels {
+
+template <int ndim>
+void debug_print(TensorListView<StorageCPU, uint8_t, ndim> tlv, int height, int width) {
+  for (int y = 0; y < height; y++) {
+    for (int x = 0; x < width; x++) {
+      std::cout << std::setw(2) << int(*tlv[0](y, x)) % 100;
+    }
+    std::cout << endl;
+  }
+}
+
+void verify_regions(ivec<4> region_shape, ivec<4> sample_shape, ivec<4> expected_cover) {
+  auto cover = div_ceil(sample_shape, region_shape);
+  EXPECT_EQ(cover, expected_cover);
+  int idx = 0;
+  for (int d0 = 0; d0 < expected_cover[0]; d0++) {
+    for (int d1 = 0; d1 < expected_cover[1]; d1++) {
+      for (int d2 = 0; d2 < expected_cover[2]; d2++) {
+        for (int d3 = 0; d3 < expected_cover[3]; d3++) {
+          auto regions_start = get_region_start(idx, region_shape, sample_shape);
+          ivec<4> region_start_expected = {d0 * region_shape[0], d1 * region_shape[1],
+                                           d2 * region_shape[2], d3 * region_shape[3]};
+          EXPECT_EQ(regions_start, region_start_expected);
+          idx++;
+        }
+      }
+    }
+  }
+}
+
+TEST(EraseGpuKernelTest, CheckUtils) {
+  {
+    ivec<4> region_shape = {2, 2, 32, 32};
+    ivec<4> sample_shape = {16, 8, 64, 64};
+    ivec<4> expected_cover = {8, 4, 2, 2};
+    verify_regions(region_shape, sample_shape, expected_cover);
+  }
+  {
+    ivec<4> region_shape = {2, 2, 32, 32};
+    ivec<4> sample_shape = {16, 8, 32, 64};
+    ivec<4> expected_cover = {8, 4, 1, 2};
+    verify_regions(region_shape, sample_shape, expected_cover);
+  }
+  {
+    ivec<4> region_shape = {2, 2, 32, 32};
+    ivec<4> sample_shape = {16, 8, 64, 32};
+    ivec<4> expected_cover = {8, 4, 2, 1};
+    verify_regions(region_shape, sample_shape, expected_cover);
+  }
+  {
+    ivec<4> region_shape = {2, 2, 32, 32};
+    ivec<4> sample_shape = {16, 8, 32, 32};
+    ivec<4> expected_cover = {8, 4, 1, 1};
+    verify_regions(region_shape, sample_shape, expected_cover);
+  }
+}
+
+template <int ndim>
+struct EraseTestParams {
+  int num_erase_regions;
+  int region_generation;
+  int fill_type;
```

Comment:
Shouldn't those two be enums? Or at least you could compare them against enum values instead of explaining the meaning in comments.